invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/fields/Identifiers/PIDField/components/OptionalDOIoptions.js

```diff
@@ -23,24 +24,39 @@ export class OptionalDOIoptions extends Component {
   };
 
   _render = (cmp, shouldWrapPopup, message) =>
-    shouldWrapPopup ? <Popup content={message} trigger={cmp} /> : cmp;
+    shouldWrapPopup && message ? <Popup content={message} trigger={cmp} /> : cmp;
 
   render() {
-    const { isManagedSelected, isNoNeedSelected, pidLabel, optionalDOItransitions } =
-      this.props;
-
-    const allDOIoptionsAllowed = _isEmpty(optionalDOItransitions);
-    const isUnManagedDisabled =
-      isManagedSelected ||
-      (!allDOIoptionsAllowed &&
-        !optionalDOItransitions.allowed_providers.includes("external"));
-    const isNoNeedDisabled =
-      isManagedSelected ||
-      (!allDOIoptionsAllowed &&
-        !optionalDOItransitions.allowed_providers.includes("not_needed"));
+    const {
+      isManagedSelected,
+      isNoNeedSelected,
+      pidLabel,
+      optionalDOItransitions,
+      record,
+    } = this.props;
+
+    const doi = record?.pids?.doi?.identifier || "";
```

Comment:
minor: see my comment below(can be replacing empty string) why the empty string is needed in the first place?

Reply:
The empty string is the initial state when the form is rendered. This was carried over from the existing implementation and I didn't refactor it. If you don't think it is a major blocker I would avoid refactoring more now...